### PR TITLE
add longhorn image support

### PR DESCRIPTION
**Pull request by mahmut-Abi**
### Describe what this PR does / why we need it

**what**

​	This PR is a longhorn image witch can provide a serries storage method.

**why**

​	Longhorn can provide a lightweight, reliable and easy-to-use distributed block storage system for Kubernetes.

### Does this pull request fix one issue?

Add support for longhorn. ( issuecomment-986387008)

#607

### Describe how you did it

1. run a sealer kubernetes:v1.19.9 images cluster.
2. Install longhorn in the cluster by using helm.
3. Get the longhorn.yaml manifest file.
4. Add Kubefile
5. Run ```sealer build -f Kubefile -t longhorn .  ```

### Describe how to verify it

1. Run  the longhorn image with sealer.
2. wait all pods are up status
3. add a nginx pod with a PVC with longhorn storageclass.
4. Run ```kubectl  get svc -n longhorn-system```  to get ```longhorn-frontend  ``` svc cluster-ip, and browser it when the nginx pod is up.
5. We can see the nginx's log storaged in longhorn.

### Special notes for reviews

NONE

### CODE REVIEW

File: applications/longhorn/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Once Longhorn has been installed in your Kubernetes cluster, you can access the UI dashboard.
+
+1. Get the Longhorn’s external service IP:
+
+​		``` kubectl -n longhorn-system get svc```
+
+​		For Longhorn v0.8.0, the output should look like this, and the `CLUSTER-IP` of the `longhorn-frontend` is used to access the Longhorn UI:
+
+2. Navigate to the IP of `longhorn-frontend` in your browser.
+
```
Comment (fanux) [inline applications/longhorn/README.md:33]: Awesome work~ could you delete the redundant blanks~

Comment (mahmut-Abi) [reply applications/longhorn/README.md:33]: ok, I will do it soon. thanks.

File: applications/longhorn/Kubefile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+FROM kubernetes:v1.19.9
```
Comment (kakaZhou719) [inline applications/longhorn/Kubefile:1]: hi @mahmut-Abi ,good job, here i suggest we use "kubernetes:v1.19.8" as base image. "kubernetes:v1.19.8"  has passed fully test by our team.

Comment (mahmut-Abi) [reply applications/longhorn/Kubefile:1]: OK, I'll notice. I'll mention commit again this time. I'll change it and upload it in the future. Don't bother you. Thank you.

**Comment by fanux** [comment]
Could you push the CloudImage in a public registry and link it in the docs?

**Comment by fanux** [comment]
Add the CloudImage in image list : https://github.com/alibaba/sealer/tree/main/applications#cloud-image-list : )

**Comment by mahmut-Abi** [comment]
excuse me please, I wonder how to login the sealer, I have tried login with alibabacloud account, and AK SK, but they did not work.&nbsp; I have tried push my images to "registry.cn-qingdao.aliyuncs.com/sealer-io/"&nbsp; &amp;&nbsp; "registry.cn-qingdao.aliyuncs.com/sealer-io/", but both of them required me to login.
can you tell&nbsp; me how to login OR what account can to login the registry.
thanks.








麦合木提·买买提
系统研发工程师
统信软件


 联系我









&nbsp;
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;;
Date: &nbsp;Mon, Dec 27, 2021 11:41 AM
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: &nbsp;Re: [alibaba/sealer] add longhorn image support (PR #961)

&nbsp;



 
Could you push the CloudImage in a public registry and link it in the docs?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fanux** [comment]
You don't need to login to registry.cn-qingdao.aliyuncs.com, just push it to your own public registry like docker hub or ACR as third part CloudImage, official registry will sync it after full test.

**Comment by fanux** [comment]
Our dingtalk group: 34619594

**Review by fanux — APPROVED** [review]

**Comment by fanux** [comment]
You can add CloudImage in image list later:  https://github.com/alibaba/sealer/tree/main/applications#cloud-image-list

**Pull request merged**